sql/2021/http/http2_3rd_party_by_types.sql

```diff
@@ -0,0 +1,40 @@
+# standardSQL
+# HTTP/2 3rd Party by Types
+SELECT
+  percentile,
+  client,
+  category,
+  APPROX_QUANTILES(http2_3_pct, 1000)[OFFSET(percentile * 10)] AS http2_3_pct
+FROM (
+  SELECT
+    client,
+    page,
+    category,
+    COUNTIF(LOWER(http_version) IN ('http/2', 'http/3', 'quic', 'h3-29', 'h3-q050')) / COUNT(0) AS http2_3_pct
+  FROM (
+    SELECT
+      client,
+      page,
+      url,
+      category,
+      protocol AS http_version
+    FROM
+      `httparchive.almanac.requests` r,
+      `httparchive.almanac.third_parties` tp
```

Comment:
These tables are cross-joined together. Should they be a plain `JOIN` with `USING`? I'd expect that to use a lot less processing.

Reply:
I can use a `USING` as the column names don't match for the domain:

```sql
    WHERE
      r.date = '2021-07-01' AND
      tp.date = '2021-07-01' AND
      NET.HOST(url) = domain)
```

I guess I could use a subquery to normalise the names (but then is that really any more efficient?)

Or I could use `ON`:

```sql
ON r.date = tp.date AND NET.HOST(url) = domain
    WHERE
      r.date = '2021-07-01' AND
      tp.date = '2021-07-01'
      )
```

but I presume that's the same as a cross join anyway?

Either way this query uses 210.5 GB and takes 17 seconds to process so small in terms of BigQuery so don't think a big deal.

Reply:
Fair enough!